hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/HollowScreening.java

```diff
@@ -0,0 +1,180 @@
+        this.reqOtherPartyKeys = otherPartyKeys;
+        this.sigMetaLookup = sigMetadataLookup;
+        this.hollowFinalizations = new HashMap<>();
+        this.reqOtherPartyKeysDeHollowed = null;
+    }
+
+    // add javadocs
+    public void execute() {
+        final var hollowKeysInOtherReqKeysIndex = createHollowKeyIndex();
+        for (final var sig : txnSigs) {
+            if (sig.getSignatureType() != SignatureType.ECDSA_SECP256K1) {
+                continue;
+            }
+            final var expandedPublicKeyDirect = sig.getExpandedPublicKeyDirect();
+            final var evmAddress = MiscCryptoUtils.extractEvmAddressFromDecompressedECDSAKey(expandedPublicKeyDirect);
+            final var compressedSecp256k1 = MiscCryptoUtils.compressSecp256k1(expandedPublicKeyDirect);
+            final var key = new JECDSASecp256k1Key(compressedSecp256k1);
+            // replace any JHollowKeys in the req keys that are derived from this active ECDSA key
+            // so that subsequent key activation verifications pass
+            replaceHollowKeysInReqKeysIfNeeded(hollowKeysInOtherReqKeysIndex, evmAddress, key);
+            // check if the evm address is linked to a hollow account and add it to finalizations map
+            addToHollowFinalizationsIfNeeded(evmAddress, key);
+        }
+    }
+
+    public boolean hasReplacedHollowKeysInOtherReqKeys() {
+        return reqOtherPartyKeysDeHollowed != null;
+    }
+
+    public JKey getScreenedPayerKey() {
+        return reqPayerKey;
+    }
+
+    public List<JKey> getOtherPartyKeysDeHollowed() {
+        return reqOtherPartyKeysDeHollowed;
+    }
+
+    public Map<byte[], JKey> getHollowFinalizations() {
+        return hollowFinalizations;
+    }
+
+    private void addToHollowFinalizationsIfNeeded(final byte[] evmAddress, final JECDSASecp256k1Key key) {
+        final var alias = ByteStringUtils.wrapUnsafely(evmAddress);
+        final var aliasedID = AccountID.newBuilder().setAlias(alias).build();
+        final var lookupResult = sigMetaLookup.aliasableAccountSigningMetaFor(aliasedID, linkedRefs);
+        if (!lookupResult.succeeded()) {
+            // in pre-handle context, there may be a hollow account with this evmAddress in the working state, but
+            // `SigMetaDataLookUp`
+            // is not aware of it; or there is a pending hollow account creation txn not yet executed;
+            // in any case, SigMetadataLookup already would have linked the evm address alias and (possibly) account ID
+            // so here link the ECDSA public key alias as well just in case
+            linkPublicKeyAliasIfNeeded(key);
+            return;
+        }
+        final var metadata = lookupResult.metadata();
+        final var accountKey = metadata.key();
+        if (accountKey.hasHollowKey()) {
+            // we have a matched hollow account, so this is a potential finalization;
+            hollowFinalizations.putIfAbsent(evmAddress, key);
```

Comment:
If we inject `AliasManager` in `Expansion` and `Rationalization`, we could directly save the `EntityNum`s of the hollow accounts here in the `hollowFinalizations` map in the `TxnAccessor`, thus getting rid of the alias resolving in handle() in `HollowFinalizationLogic`. 🤔